RecipeMatch/View Controllers/PreferencesViewController.m

```diff
@@ -45,29 +51,59 @@ - (void)viewDidLoad {
+        [self.preferencesDict setObject:self.caloriesLabel.text forKey:CALORIES_KEY];
+    }
+    [delegate sendPreferences:self.preferencesDict];
+    [self.navigationController popViewControllerAnimated:YES]; // go back to stream VC
+}
+
+// clears all preferences and resets dropdown menu titles
+- (IBAction)didTapClearPreferences:(id)sender {
+    [self.preferencesDict removeAllObjects];
+    [self setupView];
+    self.caloriesLabel.text = CALORIES_DEFAULT;
+    [self.slider setValue:[CALORIES_DEFAULT floatValue]];
+}
+
+- (IBAction) sliderValueChanged:(UISlider *)sender {
+    self.caloriesLabel.text = [NSString stringWithFormat:@"%.0f", [sender value]];
 }
 
 - (void)setupView {
     CGRect frame = CGRectMake((CGRectGetWidth(self.view.frame)-DROPDOWN_X_OFFSET), DROPDOWN_Y_POS, DROPDOWN_WIDTH, DROPDOWN_HEIGHT);
-    self.cuisineMenu = [self getDropDownMenuWithFrame:frame items:@[@"American", @"Asian", @"British",@"Caribbean",@"Central Europe",@"Chinese", @"Eastern Europe", @"French", @"Indian", @"Italian", @"Japanese", @"Kosher", @"Mediterranean", @"Mexican", @"Middle Eastern", @"Nordic", @"South American", @"South East Asian", @"no preference"]];
+    self.cuisineMenu = [self getDropDownMenuWithFrame:frame key:CUISINE_KEY items:@[@"American", @"Asian", @"British",@"Caribbean",@"Central Europe",@"Chinese", @"Eastern Europe", @"French", @"Indian", @"Italian", @"Japanese", @"Kosher", @"Mediterranean", @"Mexican", @"Middle Eastern", @"Nordic", @"South American", @"South East Asian", @"no preference"]];
     [self.view addSubview:self.cuisineMenu];
 
     frame = CGRectOffset(frame, 0, MENU_OFFSET);
-    self.healthMenu = [self getDropDownMenuWithFrame:frame items:@[@"vegan", @"vegetarian", @"tree-nut-free",@"low-sugar",@"shellfish-free",@"pescatarian", @"paleo", @"gluten-free", @"fodmap-free", @"no preference"]];
+    self.healthMenu = [self getDropDownMenuWithFrame:frame key:HEALTH_KEY items:@[@"vegan", @"vegetarian", @"tree-nut-free",@"low-sugar",@"shellfish-free",@"pescatarian", @"paleo", @"gluten-free", @"fodmap-free", @"no preference"]];
     [self.view addSubview:self.healthMenu];
 
     frame = CGRectOffset(frame, 0, MENU_OFFSET);
-    self.dietMenu = [self getDropDownMenuWithFrame:frame items:@[@"balanced", @"high-fiber", @"high-protein",@"low-carb",@"low-fat",@"low-sodium",@"no preference"]];
+    self.dietMenu = [self getDropDownMenuWithFrame:frame key:DIET_KEY items:@[@"balanced", @"high-fiber", @"high-protein",@"low-carb",@"low-fat",@"low-sodium",@"no preference"]];
     [self.view addSubview:self.dietMenu];
     
     frame = CGRectOffset(frame, 0, MENU_OFFSET);
-    self.mealMenu = [self getDropDownMenuWithFrame:frame items:@[@"breakfast", @"dinner", @"lunch",@"snack",@"teatime",@"no preference"]];
+    self.mealMenu = [self getDropDownMenuWithFrame:frame key:MEAL_TYPE_KEY  items:@[@"breakfast", @"dinner", @"lunch",@"snack",@"teatime",@"no preference"]];
     [self.view addSubview:self.mealMenu];
 }
```

Comment:
If you want to take it to a step further.....
then you can add enums for the different type of menus, and then call `getDropDownMenu` only with frame and `type`, then add a helper method that receives `type` and returns the items array 
😀